src/ed_py/world_model.py

```diff
@@ -0,0 +1,244 @@
+
+# noinspection PyUnresolvedReferences
+import tf2_geometry_msgs
+
+# noinspection PyUnresolvedReferences
+import tf2_pykdl_ros
+
+from ed_msgs.srv import Configure, SimpleQuery, SimpleQueryRequest, UpdateSrv, Reset
+
+from .entity import Entity, from_entity_info
+
+
+class WM:
+    def __init__(self, tf_buffer=None, ns: str = None):
+        if tf_buffer is None:
+            self.tf_buffer = tf2_ros.Buffer()
+            self._tf_listener = tf2_ros.TransformListener(self.tf_buffer)
+        else:
+            self.tf_buffer = tf_buffer
+
+        if ns is None:
+            prefix = ""
+        else:
+            prefix = f"/{ns}/"
+
+        self._ed_simple_query_srv = self.create_service_client(f"{prefix}ed/simple_query", SimpleQuery)
+        self._ed_update_srv = self.create_service_client(f"{prefix}ed/update", UpdateSrv)
+
+        self._ed_configure_srv = self.create_service_client(f"{prefix}ed/configure", Configure)
+        self._ed_reset_srv = self.create_service_client(f"{prefix}ed/reset", Reset)
+
+    @staticmethod
+    def create_service_client(name: str, srv_type):
+        """
+        Creates a service client
+
+        :param name: string with the name of the service in the correct namespace
+        :param srv_type: service type
+        :return: the service client
+        """
+        return rospy.ServiceProxy(name, srv_type)
+
+    def get_entities(
+        self,
+        center_point: VectorStamped,
+        etype: str = "",
+        radius: float = float("inf"),
+        uuid: str = "",
+        ignore_z: bool = False,
+    ) -> List[Entity]:
+        """
+        Get entities via Simple Query interface
+
+        :param etype: Type of entity
+        :param center_point: Point from which radius is measured
+        :param radius: Distance between center_point and entity
+        :param uuid: uuid of entity
+        :param ignore_z: Consider only the distance in the X,Y plane for the radius from center_point
+        """
+        center_point_in_map = self.tf_buffer.transform(center_point, "map")
+        query = SimpleQueryRequest(
+            id=uuid,
+            type=etype,
+            center_point=tf2_ros.convert(center_point_in_map, PointStamped).point,
+            radius=radius,
+            ignore_z=ignore_z,
+        )
+
+        try:
+            entity_infos = self._ed_simple_query_srv(query).entities
+            entities = list(map(from_entity_info, entity_infos))
+        except Exception as e:
+            rospy.logerr(
+                f"get_entities(uuid={uuid}, type={etype}, center_point={center_point}, radius={radius}, "
+                f"ignore_z={ignore_z})\n{e}, "
+                f"{traceback.format_exc()}"
+            )
+            return []
+
+        return entities
+
+    def get_closest_entity(self, center_point: VectorStamped, etype: str = "", radius: float = float("inf")) -> Entity:
+
+        entities = self.get_entities(center_point=center_point, etype=etype, radius=radius)
```

Comment:
What entities do you get when you leave etype the default value?

Reply:
If no type is specified, all entities that satisfy the other constrains are returned